src/sorting/sorting.py

```diff
@@ -1,28 +1,73 @@
 # TO-DO: complete the helper function below to merge 2 sorted arrays
-def merge(arrA, arrB):
+def helper_merge(arrA, arrB):
     elements = len(arrA) + len(arrB)
-    merged_arr = [0] * elements
+    merged_arr = [] * elements
 
     # Your code here
-
+    a = b = 0
+    while a < len(arrA) and b < len(arrB):
+        if arrA[a] < arrB[b]:
+            merged_arr.append(arrA[a])
+            a += 1
+        else:
+            merged_arr.append(arrB[b])
+            b += 1
+    merged_arr += arrA[a:]
+    merged_arr += arrB[b:]
```

Comment:
Confession time: Merge sort is my absolute favorite sorting algorithm to implement. It just makes me happy.

Ok, with that confession out of the way, you absolutely _killed_ this. 

I have only a few minor notes, and they're mostly nitpicky / related in some ways to code it appears you were semi-provided

#### 1: You're adding `elements` to memory and don't need to

I think those first two lines are attempting to create a `merged_arr` that is the appropriate length to make any updates to the array `O(1)` since you're just swapping an already indexed value instead of appending to an empty list which can occasionally be an `O(n)` operation (though the amortized cost is much better, usually).

My only note on this one is that `elements` doesn't need to exist. If you _are_ trying to do the above in those first two lines of the function, you can inline it like so:

```python
merged_arr = [] * (len(arrA) + len(arrB))
```

#### 2: Line 4 isn't doing anything at all.

As I said above, it seems that what you're attempting to do in lines 3 & 4 is micro-optimize your work with the list to ensure that you have a list that is the appropriate length, but filled with 0s. Something akin to this:

```python
>>> [0] * (4 + 4)
[0, 0, 0, 0, 0, 0, 0, 0]
```

This can definitely have some benefits in certain implementations. However, in this particular case line 4 isn't doing anything because there's nothing inside of the brackets. So what's _actually_ happening is this:

```python
>>> [] * (4 + 4)
[]
```

The rest of your code is using append and the `a`/`b` pointer values, so I actually thing the right thing to do here is to delete line 3 and replace line 4 with this:

```python
merged_arr = []
```

Has the same effect as what you're currently doing, and thanks to the amortized cost of appending it doesn't cost you that much in the real world.

#### 3: Use `continue` whenever possible to reduce complexity

`break` and `continue` are your best friends in a while loop when it comes to both denesting and improving the readability of your code. If you use `continue in your first `if` block after `a += 1`, you can actually remove the `else` expression and dedent lines 13 and 14, giving you this:

```python
def helper_merge(arrA, arrB):
    merged_arr = []

    a = b = 0
    while a < len(arrA) and b < len(arrB):
        if arrA[a] < arrB[b]:
            merged_arr.append(arrA[a])
            a += 1
            continue

        merged_arr.append(arrB[b])
        b += 1

    merged_arr += arrA[a:]
    merged_arr += arrB[b:]
``` 

#### 4: use the splat operator to combine lists

This is a newer feature, and therefore less well known, but it's super great and will make total sense coming from JavaScript.

To combine your arrays -- including any left over values in `arrA` or `arrB` -- at the end, you can actually do this:

```python
merged_arr += [*arrA[a:], *arrB[b:]]
```

Using the splat operators this way would be the equivalent in JS of doing this with the spread operator:

```javascript
[ ...mergedArr, ...arrA.slice(a), ...arrB.slice(b)]
```

This gives you a final function that looks like this:

```python
def helper_merge(arrA, arrB):
    merged_arr = []
    a = b = 0

    while a < len(arrA) and b < len(arrB):
        if arrA[a] < arrB[b]:
            merged_arr.append(arrA[a])
            a += 1
            continue

        merged_arr.append(arrB[b])
        b += 1

    merged_arr += [*arrA[a:], *arrB[b:]]
    return merged_arr
```